trl/trainer/grpo_trainer.py

```diff
@@ -1278,17 +1432,17 @@ def _generate_and_score_completions(
                 torch.no_grad(),
                 FSDP.summon_full_params(self.model_wrapped, recurse=False) if self.is_fsdp_enabled else nullcontext(),
             ):
+                prompt_inputs["input_ids"], prompt_inputs["attention_mask"] = prompt_ids, prompt_mask
                 prompt_completion_ids = unwrapped_model.generate(
-                    prompt_ids, attention_mask=prompt_mask, generation_config=self.generation_config
+                    **prompt_inputs, generation_config=self.generation_config
```

Comment:
FYI, because of this, the `fake_generate` in the test now expects the argument to have `input_ids` rather than previously, where it didn't matter